### ISSUE TITLE
The application ... can’t be opened (MacOS)

### ISSUE THREAD

**Issue by CryptoGazer** [flet-dev/flet#2265]
I'm creating a Flet desktop application and every time I want to open it on another device or in another folder on my Mac, I get the error: The application <name> can’t be opened. Previously I've packed the app using `flet pack` and the .app file and the executable were saved to the "dist" folder. However if I move my .app file to another folder or my dist folder to another location or open my file on another computer, I cannot open it!

How could I solve this issue?

Flet version: 0.17.0

**Operating system**:

OS: macOS Ventura Version 13.5

**Comment by Hornbag** [comment]
Apple imposes strict requirements for packaged application bundles, and if these requirements are not met, it will block the execution of your app on any other machine you try to run it on. There is no way to override this.

At minimum, applications need to be code signed using an Apple Developer ID (yes, you need to pay Apple money to run your own apps, even ones for private use). You can provide this using the `--codesign-identity` option in the `flet pack` command.

Additionally, Apple encourages (and may require for new OS) having your software notarised, which is an automated malware checking process. 

More info here: https://support.apple.com/en-au/guide/security/sec3ad8e6e53/web

TL;DR: Give Apple money so that it stops pretending to care about preventing malware

**Comment by FeodorFitsner** [comment]
Closing in favour of https://github.com/flet-dev/flet/issues/2347
